### PR TITLE
Topic-recognition-prostate-segment-3Dunet

### PR DESCRIPTION
**Pull request by Brady-Zhang**
implement 3D unet to segment 3D MR image of prostate

### CODE REVIEW

**Comment by WeiDaiwow** [comment]
# Document

- Title: 
	Good
- Address problem:
	Good
- Working Principle:
	Not mentioned. Description of Unet should be provided
- Figure visualization:
	No figures of UNET. Does have pipeline figure
- dependency:
	Good
- inputs and outputs description:
	Not provided. No visualization of segmented output and no dsc metric table
- training plot:
	Good 
- readme using markdown:
	Good
- preprocessing description:
	Good
- dataset split description
	Good
- Description of usage:
	Usage of the files provided. However, no description for them.


# Check the files:
- modules.py
	Good
- dataset.py		
	More comments should be provided within loading function
- train.py
	Single class DSC was used instead of multi-class
	More comments should be provided within the training function 
	Using testing images as validation? Could not find validation set
- predict.py
	DSC score calculation does not seem right. 
	no visualization function for outputs
	
Minor tips (nothing to do with marking)
	


# Overall:

- solve the problem:
	Not sure. No output visualization provided
- TF/torch implementation:
	Yes
- Design quality:
	Code looks good, not sure about the performance
- code comments:
	Only brief comments provided
- Difficulty: 
	Unet3D - Normal

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem (no output visualised, unknown what results are) -2
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage 
Module present
Commenting
No Data leakage
Difficulty: Normal -5
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe OK
Model/technical explanation -2
Good Description and Comments but could be better -1
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description minimal -1
